sisl/physics/hamiltonian.py

```diff
@@ -551,7 +551,7 @@ def _Ef(q, eig):
             Ef[1] = _Ef(q[1], bz.eigh(spin=1))
         else:
             # Ensure a single charge
-            q = q.sum()
+            q = q.sum() * 0.5 # Account for double occupancy
```

Comment:
In cases where the user supplies the correct number of electrons, then this won't work, no?

If a user supplies the number of electrons, shouldn't we expect it to be correct, or?

Reply:
Also true, although if the user supplies `q = [q[0], q-q[0]]`, and the Hamiltonian is not spin polarized, then `q = N` and we have the same problem as above, no?

Reply:
Agreed, but then it is a question of what we expect the user to input. Should they always input the # of electrons based on spin, or should it be the number of electrons in terms of the basis functions...

Reply:
OK, I think I wasn't seeing the case where a list of `q=[q0, q1]` is passed but the Hamiltonian is non-spin-polarized as something that would make sense... But I guess that for some calculations this might be the situation?